pkg/tmpl/context_funcs.go

```diff
@@ -217,27 +232,35 @@ func (c *Context) ReadFile(filename string) (string, error) {
 }
 
 func (c *Context) ReadDir(path string) ([]string, error) {
+	return c.ReadFromDir(path, false)
+}
+
+func (c *Context) ListDir(path string) ([]string, error) {
+	return c.ReadFromDir(path, true)
+}
+
+func (c *Context) ReadFromDir(path string, includeDirs bool) ([]string, error) {
 	var contextPath string
 	if filepath.IsAbs(path) {
 		contextPath = path
 	} else {
 		contextPath = filepath.Join(c.basePath, path)
 	}
 
-	entries, err := os.ReadDir(contextPath)
+	entries, err := c.readDir(contextPath)
```

Comment:
why change to c.readDir?

Reply:
Because then i can mock it just like it was mocked in ReadFile function:

https://github.com/helmfile/helmfile/blob/047de0aa14bb7f02e790e9280c0ad31ce2dcb57b/pkg/tmpl/context_funcs.go#L227

Reply:
For E2E tests it is using real os.ReadDir function and for tests it is using the mocked one then.

Reply:
ok